Mathlib/NumberTheory/NumberField/Units/Basic.lean

```diff
@@ -154,6 +174,36 @@ theorem rootsOfUnity_eq_torsion [NumberField K] :
     exact ⟨↑(torsionOrder K), (torsionOrder K).prop, h⟩
   · exact Subtype.ext_iff.mp (@pow_card_eq_one (torsion K) _ _ ⟨ζ, h⟩)
 
+section odd
+
+variable {K}
+
+theorem torsionOrder_eq_two_of_odd_finrank [NumberField K]
```

Comment:
Surely, if you modify the above lemma as suggested, this can proved more directly.